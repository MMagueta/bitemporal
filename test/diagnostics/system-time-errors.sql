\set QUIET 'on'
\pset footer off

drop schema if exists st cascade;
create schema st;

\echo 'bitemporal.system_time_errors - [ok]'
\pset title 'st.ok'
create table st.ok( system_period int4range not null );
select * from bitemporal.system_time_errors('st.ok');

\echo 'bitemporal.system_time_errors - [missing valid time]'
\pset title 'st.missing'
create table st.missing( id int );
select * from bitemporal.system_time_errors('st.missing');

\echo 'bitemporal.system_time_errors - [nullable]'
\pset title 'st.nullable'
create table st.nullable( system_period int4range );
select * from bitemporal.system_time_errors('st.nullable');

\echo 'bitemporal.system_time_errors - [invalid type]'
\pset title 'st.invalid_type'
create table st.invalid_type( system_period int not null );
select * from bitemporal.system_time_errors('st.invalid_type');

\echo 'bitemporal.system_time_errors - [nullable invalid type]'
\pset title 'st.nullable_invalid_type'
create table st.nullable_invalid_type( system_period int );
select * from bitemporal.system_time_errors('st.nullable_invalid_type');
