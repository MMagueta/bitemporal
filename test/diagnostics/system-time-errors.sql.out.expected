             st.ok
 namespace | relation | message 
-----------+----------+---------

                        st.missing
 namespace | relation |              message              
-----------+----------+-----------------------------------
 st        | missing  | Missing attribute "system_period"

                               st.nullable
 namespace | relation |                     message                      
-----------+----------+--------------------------------------------------
 st        | nullable | Attribute "system_period" should be not nullable

                                                 st.invalid_type
 namespace |   relation   |                                       message                                        
-----------+--------------+--------------------------------------------------------------------------------------
 st        | invalid_type | Attribute "system_period" with invalid type. Expected: "int4range", found: "integer"

                                                 st.nullable_invalid_type
 namespace |       relation        |                                       message                                        
-----------+-----------------------+--------------------------------------------------------------------------------------
 st        | nullable_invalid_type | Attribute "system_period" should be not nullable
 st        | nullable_invalid_type | Attribute "system_period" with invalid type. Expected: "int4range", found: "integer"

